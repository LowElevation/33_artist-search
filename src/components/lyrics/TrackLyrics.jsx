import React from "react";
import PropTypes from "prop-types";
import uuid from "react-uuid";

const TrackLyrics = ({ lyrics }) => {
  const parsedLyrics = lyrics.split("\n").map((line) => {
    return <li key={uuid()}>{line}</li>;
  });

  return <ul>{parsedLyrics}</ul>;
};

TrackLyrics.propTypes = {
  lyrics: PropTypes.string.isRequired,
};

export default TrackLyrics;
