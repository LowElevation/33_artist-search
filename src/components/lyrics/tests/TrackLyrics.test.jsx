import React from "react";
import { render, cleanup } from "@testing-library/react";
import TrackLyrics from "../TrackLyrics";

describe("TrackLyrics component", () => {
  afterEach(() => cleanup());
  it("renders TrackLyrics", () => {
    const { asFragment } = render(<TrackLyrics lyrics={""} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
