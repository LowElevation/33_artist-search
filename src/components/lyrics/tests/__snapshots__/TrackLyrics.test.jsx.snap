// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackLyrics component renders TrackLyrics 1`] = `
<DocumentFragment>
  <ul>
    <li />
  </ul>
</DocumentFragment>
`;
